wp-admin/js/widgets/text-widgets.js

```diff
@@ -0,0 +1,197 @@
+/* global tinymce */
+/* eslint consistent-this: [ "error", "control" ] */
+wp.textWidgets = ( function( $ ) {
+	'use strict';
+
+	var component = {};
+
+	/**
+	 * Text widget control.
+	 *
+	 * @class TextWidgetControl
+	 * @constructor
+	 * @abstract
+	 */
+	component.TextWidgetControl = Backbone.View.extend({
+
+		/**
+		 * View events.
+		 *
+		 * @type {Object}
+		 */
+		events: {},
+
+		/**
+		 * Initialize.
+		 *
+		 * @param {Object}         options - Options.
+		 * @param {Backbone.Model} options.model - Model.
+		 * @param {jQuery}         options.el - Control container element.
+		 * @returns {void}
+		 */
+		initialize: function initialize( options ) {
+			var control = this;
+
+			Backbone.View.prototype.initialize.call( control, options );
+
+			if ( ! control.el ) {
+				throw new Error( 'Missing options.el' );
+			}
+		},
+
+		/**
+		 * Render template.
+		 *
+		 * @returns {void}
+		 */
+		render: function render() {
+			var control = this, changeDebounceDelay = 1000, id, editor, textarea;
+			textarea = control.$el.find( 'textarea:first' );
+			id = textarea.attr( 'id' );
+
+			// Destroy any existing editor so that it can be re-initialized after a widget-updated event.
+			if ( tinymce.get( id ) ) {
+				delete tinymce.editors[ id ];
+				tinymce.remove( '#' + id );
+			}
+
+			wp.editor.initialize( id, {
+				tinymce: {
+					wpautop: true
+				},
+				quicktags: true
+			} );
+
+			editor = window.tinymce.get( id );
+			editor.on( 'change', _.debounce( function() {
+				editor.save();
+				textarea.trigger( 'change' );
+			}, changeDebounceDelay ) );
+		}
+	});
+
+	/**
+	 * Mapping of widget ID to instances of TextWidgetControl subclasses.
+	 *
+	 * @type {Object.<string, wp.textWidgets.TextWidgetControl>}
+	 */
+	component.widgetControls = {};
+
+	/**
+	 * Handle widget being added or initialized for the first time at the widget-added event.
+	 *
+	 * @param {jQuery.Event} event - Event.
+	 * @param {jQuery}       widgetContainer - Widget container element.
+	 * @returns {void}
+	 */
+	component.handleWidgetAdded = function handleWidgetAdded( event, widgetContainer ) {
+		var widgetContent, widgetForm, idBase, widgetControl, widgetId, animatedCheckDelay = 50, widgetInside, renderWhenAnimationDone;
+		widgetForm = widgetContainer.find( '> .widget-inside > .form, > .widget-inside > form' ); // Note: '.form' appears in the customizer, whereas 'form' on the widgets admin screen.
```

Comment:
The `handleWidgetAdded` function will only be called once for every widget created. So there is no performance concerns here.